src/k8s/pkg/k8sd/api/certificates_refresh.go

```diff
@@ -151,9 +154,29 @@ func refreshCertsRunWorker(s state.State, r *http.Request, snap snap.Snap) respo
 				return fmt.Errorf("failed to generate CSR for %s: %w", csr.name, err)
 			}
 
+			pubKey, err := pkiutil.LoadRSAPublicKey(clusterConfig.Certificates.GetK8sdPublicKey())
+			if err != nil {
+				return fmt.Errorf("failed to load k8sd public key: %w", err)
+			}
+
+			// calculate sha256 sum of CSR request
+			hash := sha256.New()
+			if _, err := hash.Write([]byte(csrPEM)); err != nil {
+				return fmt.Errorf("failed to compute sha256: %w", err)
+			}
+
+			// encrypt the hash with the public cluster RSA key
+			signature, err := rsa.EncryptPKCS1v15(rand.Reader, pubKey, hash.Sum(nil))
+			if err != nil {
+				return fmt.Errorf("failed to encrypt csr signature: %w", err)
+			}
```

Comment:
Move this to it's own function that returns the signature.

Reply:
Why would you move that into a separate function - what value do you see in doing that?

Reply:
Thanks for chiming in @neoaggelos. @bschimke95, I see merit in modularizing the code but I trust you to weigh the tradeoffs and implement the solution you think is best.